relnotes/taskext.feature.md

```diff
@@ -0,0 +1,70 @@
+## Built-in task and scheduler support in DaemonApp
+
+`ocean.util.app.DaemonApp` `ocean.util.app.ext.TaskExt`
+
+It is now possible to make `DaemonApp` initialize scheduler automatically using
+standard app configuration file. To do so, it is sufficient to enable it via
+`DaemonApp` optional settings:
+
+```D
+class MyApp : DaemonApp
+{
+  this ( )
+  {
+    DaemonApp.OptionalSettings settings;
+    settings.use_task_ext = true;
+    super("name", "desc", VersionInfo.init, settings);
+  }
+```
+
+This extension is disabled by default to avoid clashing with existing
+application code which is already handling scheduler initialization in some way.
+
+If `TaskExt` is enabled, overriden `run` method will be called within a task
+context and there is no need to create one manually or start scheduler event
+loop:
```

Comment:
I didn't realize this was under development, but just to raise a few key concerns here:

* having `run` automatically run inside a task means that there is no longer any reliable way to do things in `run` before the scheduler (and hence, epoll) event loop starts.  This may have some implications in terms of separation of concerns between when the app's data structures and functionality are set up, versus when they start _doing_ something.

* having `run` work within a task makes changes to how exceptions thrown from within it are handled, even more so if (for example) suspend and resume are involved.

Given that it's (for now) an optional change, I dare say we'll work these things out.  But both these things could have some implications for app transitions to use this new functionality.